experimental/dds/stack/src/sharedStack.ts

```diff
@@ -0,0 +1,162 @@
+import { IFluidSerializer, SharedObject } from "@fluidframework/shared-object-base";
+
+/**
+ * Events useful for testing and debugging a {@link SharedStack}
+ */
+export interface DiagnosticEvents extends IEvent {
+    (event: "downloadedBlob", handler: () => void);
+    (event: "uploadedBlob", handler: () => void);
+    (event: "processedOp", handler: () => void);
+}
+
+/**
+ * A stack DDS which stores every element in its own attachment blob.
+ */
+export class SharedStack<T = unknown> extends SharedObject<DiagnosticEvents> {
+    private head?: IFluidHandle<ArrayBufferLike>;
+
+    /** Construct a new SharedStack */
+    public constructor(
+        id: string,
+        runtime: IFluidDataStoreRuntime,
+        attributes: IChannelAttributes,
+        telemetryContextPrefix: string
+    ) {
+        super(id, runtime, attributes, telemetryContextPrefix);
+    }
+
+    /** True iff there are no elements pushed to this stack */
+    public isEmpty(): boolean {
+        return this.head === undefined;
+    }
+
+    /** Download, remove and return the top element on this stack */
+    public async pop(): Promise<T> {
+        if (this.head === undefined) {
+            throw new Error("SharedStack is empty");
+        }
+
+        const head = await get<Node<T>>(this.head, this.serializer);
+        this.emit("downloadedBlob");
+        this.setHead(head.next);
+        return head.content;
+    }
+
+    /** Upload a new element and add it to the top of the stack */
+    public async push(content: T): Promise<void> {
+        const head: Node<T> = {
+            content,
+            next: this.head
+        };
+
+        this.setHead(await this.uploadNode(head));
+    }
+
+    protected summarizeCore(serializer: IFluidSerializer, telemetryContext?: ITelemetryContext): ISummaryTreeWithStats {
+        const builder = new SummaryTreeBuilder();
+        if (this.head !== undefined) {
+            builder.addBlob("head", serializer.stringify(this.head, this.handle));
+        }
+        return builder.getSummaryTree();
+    }
+
+    protected async loadCore(services: IChannelStorageService): Promise<void> {
+        if (await services.contains("head")) {
+            const string = IsoBuffer.from(await services.readBlob("head")).toString();
+            this.head = this.serializer.parse(string);
+        }
+    }
+
+    protected processCore(message: ISequencedDocumentMessage, local: boolean, localOpMetadata: unknown) {
+        const serializedHead = message.contents as string | null;
+        this.head = serializedHead === null ? undefined : this.serializer.parse(serializedHead);
+        this.emit("processedOp");
+    }
+
+    public getGCData(fullGC?: boolean | undefined): IGarbageCollectionData {
+        // TODO:#1475: Implement GC as best as currently possible
+        return super.getGCData(fullGC);
+    }
+
+    protected onDisconnect() {}
+
+    protected applyStashedOp(content: any): unknown {
+        return;
```

Comment:
:D do we have to implement this at all?

Reply:
Of course; it's abstract in the base class. It would be a crime to implement a DDS without this feature :)